SingleSource/UnitTests/Vectorizer/VPlanNativePath/outer-loop-vect.cpp

```diff
@@ -0,0 +1,139 @@
+#include <cstddef>
+#include <cstdint>
+#include <iostream>
+#include <memory>
+
+#include "../common.h"
+
+// Tests for outer-loop vectorization in LLVM's VPlan-native path.
+
+#define DEFINE_SCALAR_AND_VECTOR_FN_FOR_OLV(Args, Loop)                       \
+  auto ScalarFn = [] Args {                                                   \
+    _Pragma("clang loop vectorize(disable) interleave_count(1)") Loop         \
+  };                                                                          \
+  auto VectorFn = [] Args {                                                   \
+    _Pragma("clang loop vectorize(enable)") Loop                              \
+  };
+
+#define DEFINE_SCALAR_AND_VECTOR_FN_FOR_NESTED_OLV(Args, Loop)                \
+  auto ScalarFn = [] Args {                                                   \
+    for (size_t i = 0; i < N; i++) {                                          \
+      _Pragma("clang loop vectorize(disable) interleave_count(1)") Loop       \
+    }                                                                         \
+  };                                                                          \
+  auto VectorFn = [] Args {                                                   \
+    for (size_t i = 0; i < N; i++) {                                          \
+      _Pragma("clang loop vectorize(enable)") Loop                            \
+    }                                                                         \
+  };
+
+int main() {
+  rng = std::mt19937(15);
+
+  {
+    // A matrix-multiplication where the second loop of the triple loop nest
+    // is vectorized (the macro adds the outer-most loop).
+    DEFINE_SCALAR_AND_VECTOR_FN_FOR_NESTED_OLV(
+      (size_t N, size_t M, size_t L,
+       int32_t *__restrict__ A, const int32_t *B, const int32_t *C),
+      for (size_t j = 0; j < L; j++) {
```

Comment:
Would be good to consistently use upper case for variable names, as per LLVM coding style.

Reply:
Sorry for this, I fixed that!